### PR TITLE
Add ExportCommand to supporting exporting of json file

### LINKED ISSUE #59: Add ExportCommand to supporting exporting of json file

### PR DESCRIPTION
**Pull request by wentingchua**
Fixes #59

### CODE REVIEW

**Review by shawnnygoh — COMMENTED** [review]
LGTM! Good job 👍🏻!

File: src/main/java/seedu/address/logic/commands/ExportCommand.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    public static final String COMMAND_WORD = "export";
+    public static final String MESSAGE_SOURCE_FILE_NOT_FOUND = "Address book not found."
+            + " Start adding contacts to form your address book.";
+    public static final String MESSAGE_INVALID_FILE_FORMAT = "Export failed. Target file must have a .json extension.";
+    public static final String MESSAGE_EXPORT_SUCCESS = "Exported Address Book to %1$s as requested ...";
+    public static final String MESSAGE_EXPORT_FAILURE = "Failed to export json file";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Exports the address book data at the specified path.\n"
+            + "Parameters: FILEPATH (must be a valid file path)\n"
+            + "Example: " + COMMAND_WORD + " C:/Users/username/Desktop/exported_data.json";
```
Comment (shawnnygoh) [inline src/main/java/seedu/address/logic/commands/ExportCommand.java:28]: Just a suggestion here: Maybe you could add the file path for Mac or Linux also!

File: src/test/java/seedu/address/logic/commands/ExportCommandTest.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        ExportCommand exportCommand1 = new ExportCommand(targetPath1);
+        ExportCommand exportCommand2 = new ExportCommand(targetPath2);
+
+        // Commands with different target paths should not be equal
+        assertFalse(exportCommand1.equals(exportCommand2));
+        assertFalse(exportCommand2.equals(exportCommand1)); // Test symmetry
+    }
+
+    @Test
+    public void equals_differentPathFormat_dependsOnPathImplementation() {
```
Comment (shawnnygoh) [inline src/test/java/seedu/address/logic/commands/ExportCommandTest.java:125]: Love the considerations given to the test cases here, keeping in mind edge cases! Keep up the good work!

File: src/test/java/seedu/address/logic/commands/ExportCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    @Test
+    public void execute_invalidFileFormat_throwsCommandException() {
+        Path invalidPath = temporaryFolder.resolve("exported_data.txt"); // Not a .json file
+        ExportCommand exportCommand = new ExportCommand(invalidPath);
+        exportCommand.sourceFileExists("");
+        CommandException exception = assertThrows(CommandException.class, () -> {
+            exportCommand.execute(model);
+        });
+        assertEquals(ExportCommand.MESSAGE_INVALID_FILE_FORMAT, exception.getMessage());
+    }
```
Comment (Meatsushi64) [inline src/test/java/seedu/address/logic/commands/ExportCommandTest.java:57]: Great job on the very comprehensive tests!

**Review by Meatsushi64 — APPROVED** [review]
Great job for the implementation of the export command and its functionalities! It would be good to explain to users in the user guide later on on how to get the correct file path that can be used by the bot!

**Review by souledfigurine — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/logic/commands/ExportCommand.java (hunk 1 above)
Comment (wentingchua) [reply src/main/java/seedu/address/logic/commands/ExportCommand.java]: Wow great catch! Thanks for pointing it out, I have added them in! 😃
